Submitted By: Tyler Berry (loki at arete dot cc)
Date: 2003-11-19
Initial Package Version: 1.4.1
Origin: http://www.geocrawler.com/mail/msg.php3?msg_id=6612801&list=709
Description: Fixes problems starting/restarting syslogd.
 
diff -Naur sysklogd-1.4.1.orig/syslogd.c sysklogd-1.4.1/syslogd.c
--- sysklogd-1.4.1.orig/syslogd.c	2001-03-11 19:40:10.000000000 +0000
+++ sysklogd-1.4.1/syslogd.c	2003-11-19 16:03:11.376442834 +0000
@@ -890,11 +890,11 @@
 		dprintf("Checking pidfile.\n");
 		if (!check_pid(PidFile))
 		{
+			signal (SIGTERM, doexit);
 			if (fork()) {
 				/*
 				 * Parent process
 				 */
-				signal (SIGTERM, doexit);
 				sleep(300);
 				/*
 				 * Not reached unless something major went wrong.  5
